src/forms/upload-google.ts

```diff
@@ -0,0 +1,140 @@
+    const postId: string = call.context.post?.id as string;
+
+    const mattermostOpts: MattermostOptions = {
+        mattermostUrl,
+        accessToken: userAccessToken,
+    };
+    const mmClient: MattermostClient = new MattermostClient(mattermostOpts);
+
+    const Post = await mmClient.getPost(postId);
+    const fileIds = Post.file_ids;
+    if (!fileIds || !fileIds.length) {
+        throwException(ExceptionType.MARKDOWN, i18nObj.__('upload-google.confirmation-call.error-upload'));
+    }
+    const fileMetadata = Post.metadata.files;
+
+    const options: AppSelectOption[] = fileMetadata.map((file) => {
+        return {
+            label: file.name,
+            value: file.id,
+        };
+    });
+
+    const fields: ExpandAppField[] = [
+        {
+            type: AppFieldTypes.STATIC_SELECT,
+            name: FilesToUpload.FILES,
+            modal_label: i18nObj.__('upload-google.confirmation-call.label-modal'),
+            options,
+            multiselect: true,
+        },
+    ];
+
+    return {
+        title: i18nObj.__('upload-google.confirmation-call.title'),
+        icon: GoogleDriveIcon,
+        fields,
+        submit: {
+            path: Routes.App.CallPathSaveFileSubmit,
+            expand: {
+                acting_user: AppExpandLevels.EXPAND_SUMMARY,
+                acting_user_access_token: AppExpandLevels.EXPAND_ALL,
+                oauth2_app: AppExpandLevels.EXPAND_ALL,
+                oauth2_user: AppExpandLevels.EXPAND_ALL,
+                post: AppExpandLevels.EXPAND_SUMMARY,
+            },
+        },
+    } as ExpandAppForm;
+}
+
+export async function uploadFileConfirmationSubmit(call: ExtendedAppCallRequest): Promise<string> {
+    const i18nObj = configureI18n(call.context);
+
+    const mattermostUrl: string = call.context.mattermost_site_url as string;
+    const botAccessToken: string = call.context.bot_access_token as string;
+    const postId: string = call.context.post?.id as string;
+    const channelId: string = call.context.post?.channel_id as string;
+    const values = call.values as SelectedUploadFilesForm;
+    const saveFiles = values.upload_file_google_drive.map((val) => val.value);
+
+    const mattermostOpts: MattermostOptions = {
+        mattermostUrl,
+        accessToken: botAccessToken,
+    };
+    const mmClient: MattermostClient = new MattermostClient(mattermostOpts);
+
+    const Post = await mmClient.getPost(postId);
+    const fileIds = Post.file_ids;
+    const filesMetadata = Post.metadata?.files;
+    const responseArray: Schema$File[] = [];
+
+    const drive = await getGoogleDriveClient(call);
+    for (let index = 0; index < fileIds.length; index++) {
+        const metadata = filesMetadata[index];
+
+        if (!saveFiles.includes(metadata.id)) {
+            continue;
+        }
+
+        const file = await mmClient.getFileUploaded(metadata.id); // eslint-disable-line no-await-in-loop
```

Comment:
Given that we implement the streaming functionality mentioned in the other comment, we would want to upload the files in parallel, so we would want to use `Promise.all` to coordinate uploading all of the files

Reply:
This was worked on PR [[feat: upload files in chunks]](https://github.com/liz-segura98/mattermost-app-google-drive/pull/3)